### DISCUSSION TITLE
Issue with graphqlGenerateSDL

### DISCUSSION THREAD

**Discussion by jgindin** [ExpediaGroup/graphql-kotlin#1274] [Q&A] [answered]
I'm trying out graphql-kotlin, primarily because of the fact that I can write the code and the SDL can be generated, which I can then (in theory) use to generate TypeScript types for my frontend.

Based on the [KTor example](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples/server/ktor-server), I've got a simple server running, using the `HelloQueryService` from that example (but I didn't take any of the other queries or mutations).

However, I'm having an issue when I try to run the graphqlGenerateSDL task:
```
$ ./gradlew graphqlGenerateSDL
> Task :compileKotlin
'compileJava' task (current target is 11) and 'compileKotlin' task (current target is 1.8) jvm target compatibility should be set to the same Java version.

> Task :compileJava NO-SOURCE
> Task :processResources
> Task :classes
> Task :graphqlGenerateSDL FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':graphqlGenerateSDL'.
> There was a failure while executing work items
   > A failure occurred while executing com.expediagroup.graphql.plugin.gradle.actions.GenerateSDLAction
      > java.util.zip.ZipException: zip END header not found
```

In my `build.gradle.kts` I have:
```
plugins {
  . . . 
  id("com.expediagroup.graphql") version "5.1.1"
  . . 
}

dependencies {
  . . . 
  implementation("com.expediagroup:graphql-kotlin-server:5.1.1")
  . . .
}

graphql {
  schema {
    packages = listOf("com.somewhere.something.graphql")
  }
}
. . . 
```

Running `gradlew` with the `--info` option, I see:
```
. . . 
> Task :graphqlGenerateSDL FAILED
file or directory '/home/me/src/something/service/build/classes/java/main', not found
Caching disabled for task ':graphqlGenerateSDL' because:
  Build cache is disabled
Task ':graphqlGenerateSDL' is not up-to-date because:
  Task has failed previously.
file or directory '/home/me/src/something/service/build/classes/java/main', not found
:graphqlGenerateSDL (Thread[Execution worker for ':' Thread 14,5,main]) completed. Took 0.121 secs.
```

I do have a `.../build/classes/kotlin/main` directory, but not `.../build/classes/java`.

Am I missing some configuration option perhaps?

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
Can you share a repo that reproduces the error?

**Reply by jgindin** [reply]
In preparation for putting a repo together, I copied my project to another computer where I plan to clean it up. However, when I ran the task on this other computer, _it works_.

So... there must be something in the environment that's causing the issue. I'm going to try to figure out what that is.

Thank you for the prompt response!

**Reply by jgindin** [reply]
Yes... it has something to do with the version of Java being used.

I'm going to close this.

Thanks again!
